iroh-dns-server/src/http.rs

```diff
@@ -35,13 +36,61 @@ use crate::{config::Config, metrics::Metrics};
 
 pub use self::tls::CertMode;
 
+/// Config for http rate limit
+#[derive(Debug, Serialize, Clone)]
+pub enum RateLimitConfig {
+    /// Enable or disable rate limiting
+    Boolean(bool),
```

Comment:
Can we just have two named cases for the two boolean values?

So
```rust
pub enum RateLimitConfig {
  Disabled,
  Enabled, // non-smart? what's the best way to describe the defaullt mode?
  Smart
}
```

Reply:
Right now `Some(Boolean(true))` and `None` have the same meaning/effect. We shouldn't create this duplicated behaviour. 

If `Boolean(true)` is kept, or _ideally_ changed to `Simple`/`Conventional`/etc then the options should go away and we should be setting the default with serde. If the option is kept then the enum should only be `Disabled` and `Smart`, but not both.

Reply:
Decided to drop Boolean and go with Disabled/Simple/Smart and leave the toml setting value as a string